godotcord_storage_manager.cpp

```diff
@@ -0,0 +1,116 @@
+#include "godotcord_storage_manager.h"
+#include "godotcord.h"
+
+#include "core/func_ref.h"
+
+GodotcordStorageManager *GodotcordStorageManager::singleton = NULL;
+
+GodotcordStorageManager* GodotcordStorageManager::get_singleton() {
+    return GodotcordStorageManager::singleton;
+}
+
+void GodotcordStorageManager::_bind_methods() {
+    ClassDB::bind_method(D_METHOD("get_path"), &GodotcordStorageManager::get_path);
+    ClassDB::bind_method(D_METHOD("read", "name"), &GodotcordStorageManager::read);
+    ClassDB::bind_method(D_METHOD("read_async", "name"), &GodotcordStorageManager::read_async);
+    ClassDB::bind_method(D_METHOD("read_async_partial", "name", "offset", "length"), &GodotcordStorageManager::read_async_partial);
+    ClassDB::bind_method(D_METHOD("write", "name", "data"), &GodotcordStorageManager::write);
+    ClassDB::bind_method(D_METHOD("write_async", "name", "data"), &GodotcordStorageManager::write_async);
+    ClassDB::bind_method(D_METHOD("destroy", "name"), &GodotcordStorageManager::destroy);
+    ClassDB::bind_method(D_METHOD("exists", "name"), &GodotcordStorageManager::exists);
+    ClassDB::bind_method(D_METHOD("stat", "name"), &GodotcordStorageManager::stat);
+    ClassDB::bind_method(D_METHOD("count"), &GodotcordStorageManager::count);
+    ClassDB::bind_method(D_METHOD("stat_at", "index"), &GodotcordStorageManager::stat_at);
+
+    ADD_SIGNAL(MethodInfo("async_data_read", PropertyInfo(Variant::POOL_BYTE_ARRAY, "data")));
+    ADD_SIGNAL(MethodInfo("async_data_written"));
+}
+
+String GodotcordStorageManager::get_path() {
+    char path[4096];
+    discord::Result result = Godotcord::get_singleton()->get_core()->StorageManager().GetPath(path);
+    ERR_FAIL_COND_V_MSG(result != discord::Result::Ok, path,"Something went wrong while getting path.")
+    return path;
+}
+
+PoolByteArray GodotcordStorageManager::read(String p_name) {
+    unsigned char* data;
```

Comment:
You have to create a uin8_t array here using `uint8_t data[length]` or `uint8_t *data = (uin8_t*)malloc(length)`(You have to free the memory when using the second option). Just creating a pointer that doesn't point anywhere won't work.

Reply:
But how should I find the length?

Reply:
Just waiting for this one.

Reply:
I'll make the changes now.